gtsam/linear/GaussianFactorGraph.h

```diff
@@ -181,19 +181,31 @@ namespace gtsam {
     ///@{
 
     /**
-     * Return vector of i, j, and s to generate an m-by-n sparse Jacobian matrix,
-     * where i(k) and j(k) are the base 0 row and column indices, s(k) a double.
+     * Return vector of i, j, and s to generate an m-by-n sparse augmented
+     * Jacobian matrix, where i(k) and j(k) are the base 0 row and column
+     * indices, s(k) a double.
      * The standard deviations are baked into A and b
+     * @return the sparse matrix as a std::vector of boost tuples
      */
     std::vector<boost::tuple<size_t, size_t, double> > sparseJacobian() const;
 
     /**
-     * Matrix version of sparseJacobian: generates a 3*m matrix with [i,j,s] entries
-     * such that S(i(k),j(k)) = s(k), which can be given to MATLAB's sparse.
+     * Matrix version of sparseJacobian: generates a 3*m matrix with [i,j,s]
+     * entries such that S(i(k),j(k)) = s(k), which can be given to MATLAB's
+     * sparse.  Note: i, j are 1-indexed.
      * The standard deviations are baked into A and b
      */
     Matrix sparseJacobian_() const;
 
+    /** Returns a sparse matrix with `int` indices instead of `size_t` for
+     * slightly faster performance
+     * @param ordering the column ordering
+     * @param[out] nrows The number of rows in the Jacobian
+     * @param[out] ncols The number of columns in the Jacobian
+     */
+    std::vector<std::tuple<int, int, double> > sparseJacobianFast(
```

Comment:
Why not just make sparseJacobian fast ?

Reply:
I was concerned about backwards compatibility, but if you think this isn't a big issue, then I would gladly make everything "fast".

API Changes:
1) If people don't use `auto` and use the actual type in declarations, e.g. `std::vector<boost::tuple<size_t, size_t, double>> jacobian = sparseJacobian()`, then obviously this will need to change
2) `boost::tuple` and `std::tuple` actually have different access methods, where `boost::tuple` uses member function `get<N>()` while `std::tuple` uses a free function `std::get<N>(tuple)` so even code that uses `auto` will break.

`boost::tuple<int, int, double>` is no faster than `boost::tuple<size_t, size_t, double>` on GCC 7-10 that I tested on Ubuntu 18.04 since it's 8 byte aligned, whereas `std::tuple` can align in 4 bytes.

Reply:
OK, now I'm confused. What *is* faster then?

Reply:
A custom tuple type will definitely be much faster (and deterministic)

Reply:
`std::tuple` is faster (`boost::tuple` is the slow one)

A custom struct was the same speed as `std::tuple`, but perhaps it might vary by compiler/architecture

Reply:
I see. Well, then, let's use std::tuple and size_t and be done with it :-) No fast or slow versions.

Reply:
Sorry, last clarification: `std::tuple<int, int, double>` is faster than `std::tuple<size_t, size_t, double>`.

| fast | slow |
|--|--|
|`std::tuple<int...>` | `std::tuple<size_t,...>`|
| custom struct | `boost::tuple<size_t, ...>`|
|  | `boost::tuple<int, ...>` |

where both "fast" options have the same speed and all 3 "slow" options have the same speed.

I will assume `std::tuple` and `int` is ok with you.  (Eigen sparse doesn't support unsigned indexing for some reason, and I don't think >2 billion entries aka 6GB worth will be needed)

Reply:
OK

Reply:
done